pkg/flow/flow.go

```diff
@@ -192,18 +184,23 @@ func newFlow(o Options) (*Flow, context.Context) {
 		sched:       sched,
 		loader:      loader,
 
-		cancel:       cancel,
 		exited:       make(chan struct{}, 1),
 		loadFinished: make(chan struct{}, 1),
-	}, ctx
+	}
+}
+
+// Starts flow controller as a go routine
+func (c *Flow) Run() {
+	go c.run()
 }
```

Comment:
IMO this is a bit of an anti-pattern, and I would rather:

1. Have `Flow.Run` be a blocking call (so the caller has to decide whether it should be launched in a goroutine rather than us making that decision for them)
2. Pass the context to use to `Flow.Run` 
3. Remove `Flow.exited` and `Flow.Stop`, which will no longer be required.

Reply:
That makes sense, I didn't know how tied we were to the existing Flow public API on this 1 so I didn't refactor it.

Reply:
Done, I just made these changes